scipy/stats/_calc_binned_statistic_pythran.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+import builtins
+
+#pythran export _create_binned_data_pythran(int64[:], int[:], float[:,:], int)
+#pythran export _create_binned_data_pythran(int64[:], int[:], int[:,:], int)
+#pythran export _create_binned_data_pythran(intc[:], int[:], float[:,:], int)
+#pythran export _create_binned_data_pythran(intc[:], int[:], int[:,:], int)
+#pythran export _create_binned_data_pythran(int[:], int[:], float[:,:], int)
+#pythran export _create_binned_data_pythran(int[:], int[:], int[:,:], int)
+def _create_binned_data_pythran(bin_numbers, unique_bin_numbers, values, vv):
```

Comment:
I think all the `_pythran` in function names can be dropped here, that would make it more readable.

In `_binned_statistic.py` you can then do:
```python
from _calc_binned_statistic_pythran.py import (
    _create_binned_data as _create_binned_data_pythran,
    _calc_binned_statistic as _calc_binned_statistic_pythran,
)
```

Reply:
Thanks! I didn't import `_create_binned_data_pythran` because it is useless in `_binned_statistic.py`. The pure Python function  `_calc_binned_statistic` should also call the pure Python  `_create_binned_data` because   `_create_binned_data` also have an argument `values` that can be object type. 

The Python version `_calc_binned_statistic` is used to handle object type `values` and callables. So actually when it is callable but `values` is not of object type, `_create_binned_data_pythran`  can be used by `_calc_binned_statistic` but the speed improvement is not large. I tried that in my first commit and you can see the benchmarks https://github.com/scipy/scipy/pull/14345#issue-683041055